ls1intum/Artemis
# `General`: Allow to restrict exercise import search to courses / exercises

# NOTE: Stacked PR based on https://github.com/ls1intum/Artemis/pull/5484

### Checklist
#### General


#### Server
#### Client
(https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context



As requested by @krusche it would be helpful if we could search in the import dialog only for course exercises or only for exam exercises. 

### Description


Added new filter parameters and corresponding queries for the filter combination possibilities. 

### Steps for Testing

Test the search functionality in the import dialog of each exercise type: Text,Quiz,Modeling,Programming:
- CourseFilter active means course exercises should be searched for.
- ExamFilter active means exam exercises should be searched for.
- Activating both -> Both exam and course exercises are searched for
- Deactivating both is disabled as it would make no sense to do so from a user perspective


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Test Coverage for critical methods:

| Class                      | Method               | Branch | Line |
|----------------------------|----------------------|--------|------|
| TextExerciseService        | getAllOnPageWithSize | 100%   | 100% |
| QuizExerciseService        | getAllOnPageWithSize | 100%   | 100% |
| ModelingExerciseService    | getAllOnPageWithSize | 100%   | 100% |
| ProgrammingExerciseService | getAllOnPageWithSize | 100%   | 100% |



### Screenshots



https://user-images.githubusercontent.com/29718932/181797266-b3c68cff-f20f-48c6-bb18-b7eb728e5f0f.mp4
